swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -199,261 +117,133 @@ simulation's `action` function.
+				os.RemoveAll(datadir)
+				store.Close()
 			}
-			rpcSubscriptionsWg.Add(1)
-			go func() {
-				<-wsDoneC
-				rpcSubscriptionsWg.Done()
-			}()
-
-			//watch for peers disconnecting
-			wdDoneC, err := streamTesting.WatchDisconnections(id, client, disconnectC, quitC)
-			if err != nil {
-				return err
-			}
-			rpcSubscriptionsWg.Add(1)
-			go func() {
-				<-wdDoneC
-				rpcSubscriptionsWg.Done()
-			}()
-		}
+			localStore := store.(*storage.LocalStore)
+			db := storage.NewDBAPI(localStore)
+			kad := network.NewKademlia(addr.Over(), network.NewKadParams())
+			delivery := NewDelivery(kad, db)
 
-		//second iteration: start syncing and setup stream subscriptions
-		for j, id := range ids {
-			log.Trace(fmt.Sprintf("Start syncing and stream subscriptions: %d", j))
-			client, err := net.GetNode(id).Client()
-			if err != nil {
-				return err
-			}
-			//start syncing!
-			var cnt int
-			err = client.CallContext(ctx, &cnt, "stream_startSyncing")
-			if err != nil {
-				return err
-			}
-			//increment the number of subscriptions we need to wait for
-			//by the count returned from startSyncing (SYNC subscriptions)
-			subscriptionCount += cnt
-			//now also add the number of RETRIEVAL_REQUEST subscriptions
-			for snid := range registries[id].peers {
-				subscriptionCount++
-				err = client.CallContext(ctx, nil, "stream_subscribeStream", snid, NewStream(swarmChunkServerStreamName, "", false), nil, Top)
-				if err != nil {
-					return err
-				}
-			}
-		}
+			r := NewRegistry(addr, delivery, db, state.NewInmemoryStore(), &RegistryOptions{
+				DoSync:          true,
+				SyncUpdateDelay: 0,
+			})
 
-		//now wait until the number of expected subscriptions has been finished
-		//`watchSubscriptionEvents` will write with a `nil` value to errc
-		//every time a `SubscriptionMsg` has been received
-		for err := range errc {
-			if err != nil {
-				return err
-			}
-			//`nil` received, decrement count
-			subscriptionCount--
-			//all subscriptions received
-			if subscriptionCount == 0 {
-				break
-			}
-		}
+			fileStore := storage.NewFileStore(storage.NewNetStore(localStore, nil), storage.NewFileStoreParams())
+			bucketKeyFileStore = simulation.BucketKey("filestore")
+			bucket.Store(bucketKeyFileStore, fileStore)
+			testRegistry := &TestRegistry{Registry: r, fileStore: fileStore}
 
-		log.Info("Stream subscriptions successfully requested, action terminated")
+			return testRegistry, cleanup, nil
 
-		if live {
-			//upload generated files to nodes
-			var hashes []storage.Address
-			var rfiles []string
-			hashes, rfiles, err = uploadFilesToNodes(nodes)
-			if err != nil {
-				return err
-			}
-			conf.hashes = append(conf.hashes, hashes...)
-			randomFiles = append(randomFiles, rfiles...)
-			//signal to the trigger loop that the upload has finished
-			uploadFinished <- struct{}{}
-		}
+		},
+	})
+	defer sim.Close()
 
-		return nil
+	log.Info("Initializing test config")
+	_, err := sim.AddNodesAndConnectFull(3)
```

Comment:
i dont understand this. Are we not connecting etc using a ssnapshot?

Reply:
this is a copy paste bug, thanks